AppController/djinn.rb

```diff
@@ -1955,6 +1955,25 @@ def update(apps, secret)
 
     # Make sure we have the latest code deployed.
     apps.each { |appid|
+      # Let's make sure the application is enabled.
+      uac = UserAppClient.new(my_node.private_ip, @@secret)
+      RETRIES.downto(0) { |tries|
+        begin
+          result = uac.enable_app(appid)
+          Djinn.log_debug("enable_app returned #{result}.")
+          break
+        rescue FailedNodeException
+          # Failed to talk to the UserAppServer: let's try again.
+          Djinn.log_debug("Failed to talk to UserAppServer for #{appid}.")
+        end
+        if tries == 0
+          Djinn.log_warn("Couldn't enable application #{appid}!")
```

Comment:
Wouldn't it be better to return a failure message to the client and prevent the AppController from trying to continue setting up the app?